### PR TITLE
Sync: Sync Setup UI

### PR DESCRIPTION
**Pull request by malmstein**
Task/Issue URL: https://app.asana.com/0/488551667048375/1205475691908272/f

### Description
This PR updates the Sync Setup UI to the latest designs

### Steps to test this PR

Scan Options is present for design purposes, but there’s no functionality

_Scan QR Code_
- [x] Open Sync Settings
- [x] Tap on Scan QR Code
- [x] Scan a code from another synced device
- [x] Verify Device Synced screen appears
- [x] Tap on Next
- [x] Verify Recovery code screen appears

_Enter Text Code_
- [x] Open Sync Settings
- [x] Tap on Enter Text Code
- [x] Enter the text code from another synced device
- [x] Verify Device Synced screen appears
- [x] Tap on Next
- [x] Verify Recovery code screen appears

_Initialize Sync_
- [x] Open Sync Settings
- [x] Tap on Inititalize Sync
- [x] Verify Sync Setup screen appears
- [x] Tap on Next
- [x] Verify Recovery code screen appears

_Recover Data_
- [x] Open Sync Settings
- [x] Tap on Recover Data
- [x] Verify Scan QR / Enter code screen appears
- [x] Scan QR or Enter code from a previously created account
- [x] Verify Device Synced screen appears
- [x] Tap on Next
- [x] Verify Recovery code screen appears

### UI changes
Design Review in Asana

### CODE REVIEW

**Comment by malmstein** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3534** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3534" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/3534?utm_source=stack-comment).

File: sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/ui/setup/SyncCreateAccountFragment.kt [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+import javax.inject.*
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+
+@InjectWith(FragmentScope::class)
+class SyncCreateAccountFragment : DuckDuckGoFragment(R.layout.fragment_sync_setup) {
+    @Inject
+    lateinit var viewModelFactory: FragmentViewModelFactory
+
+    private val binding: FragmentSyncSetupBinding by viewBinding()
```
Comment (cmonfortep) [inline sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/ui/setup/SyncCreateAccountFragment.kt:49]: NIT: update fragment layout xml name to match new class name

File: sync/sync-impl/src/main/res/layout/fragment_sync_setup.xml [hunk 2]
```diff
@@ -38,6 +38,7 @@
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
             android:padding="@dimen/keyline_4"
+            android:visibility="invisible"
```
Comment (cmonfortep) [inline sync/sync-impl/src/main/res/layout/fragment_sync_setup.xml:41]: We have code in place for this button, but it's always invisible. Maybe we can remove the code?

File: sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/ui/setup/SyncCreateAccountFragment.kt [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+    private fun processCommand(it: Command) {
+        when (it) {
+            AbortFlow -> {
+                requireActivity().setResult(Activity.RESULT_CANCELED)
+                requireActivity().finish()
+            }
+
+            FinishSetupFlow -> listener?.launchFinishSetupFlow()
+            Error -> {
```
Comment (cmonfortep) [inline sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/ui/setup/SyncCreateAccountFragment.kt:113]: Similar question. Do we need to do anything here, if not we can leave a comment to clarify is intended

File: sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/ui/setup/SetupAccountActivity.kt [hunk 4]
```diff
@@ -55,24 +50,12 @@ class SetupAccountActivity : DuckDuckGoActivity(), SetupFlowListener {
...
-
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        if (savedInstanceState != null) {
-            screen = savedInstanceState.getSerializable(SETUP_ACCOUNT_SCREEN_EXTRA) as Screen
+        screen = if (savedInstanceState != null) {
+            savedInstanceState.getSerializable(SETUP_ACCOUNT_SCREEN_EXTRA) as Screen
         } else {
-            screen = intent.getSerializableExtra(SETUP_ACCOUNT_SCREEN_EXTRA) as? Screen ?: SETUP
+            intent.getSerializableExtra(SETUP_ACCOUNT_SCREEN_EXTRA) as? Screen ?: DEVICE_SYNCED
```
Comment (cmonfortep) [inline sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/ui/setup/SetupAccountActivity.kt:58]: Previously, default value was SETUP (new: sync_setup). Have we changed the default value for anything in particular? (I would expect if no value indicated, correct value is sync_setup for this flow)

Which brings me the question, should we automatically close activity if user enters with sync_setup but signedin...could be something quick to cover this case.

File: sync/sync-impl/src/main/res/layout/activity_login_sync.xml [hunk 5]
```diff
@@ -41,7 +41,7 @@
         android:id="@+id/readTextCode"
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
-        app:primaryText="Manually Enter Code"
+        app:primaryText="@string/enter_code_screen_cta"
```
Comment (cmonfortep) [inline sync/sync-impl/src/main/res/layout/activity_login_sync.xml:44]: Nice catch 👁️

File: sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/ui/SyncActivity.kt [hunk 6]
```diff
@@ -86,17 +90,24 @@ class SyncActivity : DuckDuckGoActivity() {
...
+        EnterCodeContract(),
+    ) { resultOk ->
+        if (resultOk) {
+            viewModel.onLoginSuccess()
         }
     }
 
     private val connectFlow = registerForActivityResult(ConnectFlowContract()) { resultOk ->
-        // no-op
+        if (resultOk) {
```
Comment (cmonfortep) [inline sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/ui/SyncActivity.kt:108]: I believe this is the bit related to https://app.asana.com/0/0/1205581505958088/f

**Review by cmonfortep — APPROVED** [review]
LGTM just minor comments. New UI looks great!

I've created a few tasks in Asana, happy to merge once we clarify them

File: sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/ui/setup/SetupAccountActivity.kt (hunk 4 above)
Comment (malmstein) [reply sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/ui/setup/SetupAccountActivity.kt:58]: Discussed offline, we’ll stick with this.

**Pull request merged**